### PR TITLE
Fix panic from test_int__format__locale

### LINKED ISSUE #4588: Format panic in test_int__format__locale
## Issues

In `Lib/test/test_types.py``test_int__format__locale`, if you change from `format(x, 'n')` to `format(x, ',')` and run the test, RustPython will panic! It doesn't crash in CPython!

See #4593 for test case

### PR DESCRIPTION
**Pull request by minhrongcon2000**
Fix #4588

closes #4593

### CODE REVIEW

**Comment by youknowone** [comment]
would you remove `skip` from the test and add test case commit from #4593 to ensure it is fixed?

**Comment by minhrongcon2000** [comment]
It will still fail because I didn't integrate locale into format integer with numeric option... But if you want to remove it to test whether it is panic, I will do it!

**Comment by youknowone** [comment]
you can replace `skip` to `expectedFailure` in that case. At least we can test if it is crash or not.

File: Lib/test/test_format.py [hunk 1]
```diff
@@ -492,6 +492,14 @@ def test_precision_c_limits(self):
...
         with self.assertRaises(ValueError) as cm:
             format(c, ".%sf" % (INT_MAX + 1))
+            
+    def test_issue_4588_rustpython(self):
+        x = 123456789012345678901234567890
+        for i in range(0, 30):
+            format(x, ',')
+
+            # move to the next integer to test
+            x = x // 10
```
Comment (youknowone) [inline Lib/test/test_format.py:502]: we don't add our original tests under `Lib/`. Could you move this test under `extra_tests/snippets/`?

File: Lib/test/test_format.py [hunk 2]
```diff
@@ -493,6 +493,5 @@ def test_precision_c_limits(self):
         with self.assertRaises(ValueError) as cm:
             format(c, ".%sf" % (INT_MAX + 1))
 
-
```
Comment (youknowone) [inline Lib/test/test_format.py:496]: this is not intended change, right?

Comment (minhrongcon2000) [reply Lib/test/test_format.py:496]: Yeah! This is not intended

File: extra_tests/snippets/test_issue_4588.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+x = 123456789012345678901234567890
+for i in range(0, 30):
+    format(x, ',')
```
Comment (youknowone) [inline extra_tests/snippets/test_issue_4588.py:3]: this is test for `format`. `builtin_format.py` covers tests about it.

File: Lib/test/test_format.py (hunk 2 above)
Comment (youknowone) [reply Lib/test/test_format.py:496]: please also revert this line too

**Review by youknowone — CHANGES_REQUESTED** [review]

**Review by youknowone — APPROVED** [review]
Thank you!

**Comment by youknowone** [comment]
rerunning CI with #4593

**Comment by youknowone** [comment]
Could you check the test result? new code seems generating malformatted string.

```
---- format::tests::test_format_int_sep stdout ----
thread 'format::tests::test_format_int_sep' panicked at 'assertion failed: `(left == right)`
  left: `Ok("0000001234567,890,123,456,789,012,345,678")`,
 right: `Ok("1,234,567,890,123,456,789,012,345,678")`', common/src/format.rs:1035:9
```

**Review by youknowone — APPROVED** [review]
Thanks!

**Pull request merged**